src/main/java/com/github/manolo8/darkbot/backpage/DispatchManager.java

```diff
@@ -0,0 +1,176 @@
+package com.github.manolo8.darkbot.backpage;
+
+import com.github.manolo8.darkbot.Main;
+import com.github.manolo8.darkbot.backpage.dispatch.BiIntConsumer;
+import com.github.manolo8.darkbot.backpage.dispatch.DispatchData;
+import com.github.manolo8.darkbot.backpage.dispatch.InProgress;
+import com.github.manolo8.darkbot.backpage.dispatch.Retriever;
+import com.github.manolo8.darkbot.utils.http.Method;
+import org.intellij.lang.annotations.Language;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.BiConsumer;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class DispatchManager {
+    private final Main main;
+    private final DispatchData data;
+    private long lastDispatcherUpdate;
+
+    DispatchManager(Main main) {
+        this.main = main;
+        this.data = new DispatchData();
+    }
+
+    // TODO: remove before release, just parser testing
+    public static void main(String[] args) throws Exception {
+        System.out.println("Working Directory = " + System.getProperty("user.dir"));
+
+        String page = Files.lines(Paths.get("dispatcher.html")).collect(Collectors.joining("\n"));
+
+        DispatchData data = new DispatchData();
+
+        boolean upd = InfoReader.updateAll(page, data);
+        int i = 0;
+    }
+
+    public DispatchData getData() {
+        return data;
+    }
+
+    public boolean update(int expiryTime) {
+        try {
+            return update(main.backpage, expiryTime);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return false;
+    }
+
+    public boolean update(BackpageManager manager, int expiryTime) throws Exception {
+        if (System.currentTimeMillis() <= lastDispatcherUpdate + expiryTime) return false;
+        String page = manager.getConnection("indexInternal.es?action=internalDispatch", Method.GET).getContent();
+
+        if (page == null || page.isEmpty()) return false;
+        lastDispatcherUpdate = System.currentTimeMillis();
+        return InfoReader.updateAll(page, data);
+    }
+
+    public boolean hireRetriever(Retriever retriever) {
+        if (data.getAvailableSlots() > 0) {
```

Comment:
```suggestion
        if (data.getAvailableSlots() <= 0) return false;
```

Just to remove some indentation in the whole thing